perf(link Inventory Invoice)

lomamech opened the pull request:
perf(link Inventory Invoice)  …
- Complete translate key
- Add option in Inventory Registry for lookup invoice registry par
  Inventory item
- Add filter Option for inventory text in API for patient invoices


test(link inventory to Invoice)
- Add test E2E to verificate link between inventory to Invoice Registry

closes #2128

lomamech committed:
perf(link Inventory Invoice)
- Complete translate key
- Add option in Inventory Registry for lookup invoice registry par
  Inventory item
- Add filter Option for inventory text in API for patient invoices

lomamech committed:
test(link inventory to Invoice)
- Add test E2E to verificate link between inventory to Invoice Registry

jniles commented on client/src/i18n/fr/vouchers.json:
```diff
@@ -29,6 +29,7 @@
         "GLOBAL": {
             "CASHFLOW_OPTION": "Option pour cashflow",
             "CONVENTION_INVOICES": "Convention - Paiement factures",
+            "CONVENTION_PAYMENT_DESCRIPTION": "Paiement des Conventions: paiement de {{numInvoices}} facture(s) par {{debtorGroupName}}.",
```
:+1:

jniles commented on server/controllers/finance/patientInvoice.js:
```diff
@@ -231,6 +231,11 @@ function find(options) {
     'invoice.uuid IN (SELECT cash_item.invoice_uuid FROM cash_item WHERE cash_item.cash_uuid = ?)'
   );
 
+  filters.custom(
```
Hmm... it is probably better to do this filter on the `uuid` itself.  That way it will be super fast (the `uuid` column is INDEXed).

Something like this would be better:
```js
filters.custom('inventory_uuid', 'invoice.uuid IN (SELECT invoice_item.invoice_uuid FROM invoice_item WHERE invoice_item.inventory_uuid = ?');
```

jniles commented on client/src/modules/inventory/list/templates/action.cell.html:
```diff
@@ -14,5 +14,13 @@
+
+    <li class="divider"></li>
+    <li>
+      <a data-method="invoice" ui-sref="invoiceRegistry({ filters : { key : 'inventoryLabel', value : row.entity.label }})" href>
```
If you use the `uuid`, the query will be more efficient.  To show a nice label instead of a giant UUID, try passing in a display value like this:
```html
ui-sref="invoiceRegistry({ filters : { key : 'inventory_uuid', value : row.entity.uuid, displayValue : row.entity.label }})"
```
Then, in the invoice registry, you can bind the `displayValue` to the filter and it should filter correctly.

jniles commented on test/end-to-end/verificationLinks/verificationLinks.spec.js:
```diff
@@ -58,5 +58,14 @@ describe('Check links', () => {
     GU.expectRowCount('payment-registry', 0);
   });
 
+  it('Check link betwen Inventory Registry -> Invoice Registry', () => {
```
:+1:  This should say "between" instead of "betwen"

jniles reviewed (changes requested):
@lomamech, this is a really good start!

I'm concerned about the performance of this query.  Could you migrate this to using the `inventory_uuid`?  I've suggested how to do this.  If you need help making the label appear, you can ask me.

lomamech committed:
perf(link Inventory to invoice)
- Improve patient invoice API make the filtering by inventory with UUID
  instead Inventory.text
- Show the nice Label instead the UUID for the filtering by Inventory

jniles reviewed (approved):
bors r+

This works really well!  Well done!
